web/src/rest/publish.js

```diff
@@ -124,32 +137,22 @@ const publishRest = new Vue({
   },
 });
 
-// This has to be done with an interceptor because
-// the value of publishRest.token changes over time.
+// This is done with an interceptor because the value of
+// oauthClient.authHeaders is initialized asynchronously,
+// and doesn't exist at all if the user isn't logged in.
 // Using client.defaults.headers.common.Authorization = ...
 // would not update when the token does.
 client.interceptors.request.use((config) => {
-  if (!publishRest.token) {
+  if (!oauthClient.authHeaders) {
```

Comment:
There's no need for a conditional. If the `oauthClient` is not logged in, then `...oauthClient.authHeaders` will typically be an empty object, but that's technically an implementation detail. In other words, it's expected that downstreams treat the actual content of `oauthClient.authHeaders` as somewhat opaque, unconditionally including it in all API requests.

Also, the boolean value of an empty `Object` in JavaScript is truthy anyway, so this will never pass.